src/DataDefinitionsBundle/Resources/public/pimcore/js/export/search.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Data Definitions.
+ *
+ * LICENSE
+ *
+ * This source file is subject to the GNU General Public License version 3 (GPLv3)
+ * For the full copyright and license information, please view the LICENSE.md and gpl-3.0.txt
+ * files that are distributed with this source code.
+ *
+ * @copyright  Copyright (c) 2016-2019 w-vision AG (https://www.w-vision.ch)
+ * @license    https://github.com/w-vision/ImportDefinitions/blob/master/gpl-3.0.txt GNU General Public License version 3 (GPLv3)
+ */
+
+pimcore.object.search = Class.create(pimcore.object.search, {
+    createGrid: function ($super, fromConfig, response, settings, save) {
+        if (!Ext.ClassManager.get('Executable')) {
+            Ext.define('Executable', {
+                extend: 'Ext.data.Model',
+                fields: [
+                    {name: 'name', type: 'string'},
+                ]
+            });
+        }
+        
+        this.exportFromHere = new Ext.SplitButton({
+            text: t('data_definitions_processmanager_export_from_here'),
+            iconCls: "pimcore_icon_object pimcore_icon_overlay_add",
+            menu: []
+        });
+
+        var $this = this;
+        Ext.create('Ext.data.Store', {
+            model: 'Executable',
+            proxy: {
+                type: 'ajax',
+                url: '/admin/process_manager/executables/list-by-type',
+                extraParams: {
+                    type: 'exportdefinition'
+                },
+                reader: {
+                    type: 'json',
+                    rootProperty: 'data'
+                }
+            },
+            autoLoad: true,
+            listeners: {
+                load: function(store, executables, successful) {
+                    if (!successful) {
+                        return;
+                    }
+
+                    var exportMenu = [];
+                    executables.forEach(function (executable) {
+                        exportMenu.push({
+                            text: executable.get('name'),
+                            iconCls: "pimcore_icon_object pimcore_icon_overlay_add",
+                            handler: $this.exportObjects.bind($this, executable)
+                        });
+                    });
+
+                    if (exportMenu) {
+                        $this.exportFromHere.down("menu").add(exportMenu);
+                    }
+                }
+            }
+        });
+
+        $super(fromConfig, response, settings, save);
+        this.grid.down("toolbar").add([
+            "-",
+            this.exportFromHere,
+            "-"
+        ]);
+    },
+
+    exportObjects: function (executable, menuItem) {
+        var selected = this.grid.getSelectionModel().getSelection(), ids = [];
```

Comment:
If you select any checkbox on the listing, it will export just the items selected by the checkboxes.

If you don't select any checkboxes, it will export the entire listing (all pages) matched by the other filters.